packages/thirdweb/src/pay/buyWithFiat/getStatus.ts

```diff
@@ -180,25 +158,121 @@ export async function getBuyWithFiatStatus(
   params: GetBuyWithFiatStatusParams,
 ): Promise<BuyWithFiatStatus> {
   try {
-    const queryParams = new URLSearchParams({
-      intentId: params.intentId,
+    const result = await onrampStatus({
+      id: params.intentId,
+      client: params.client,
     });
 
-    const queryString = queryParams.toString();
-    const url = `${getPayBuyWithFiatStatusEndpoint()}?${queryString}`;
-
-    const response = await getClientFetch(params.client)(url);
-
-    if (!response.ok) {
-      const error = await response.text().catch(() => null);
-      throw new Error(
-        `HTTP error! status: ${response.status} - ${response.statusText}: ${error || "unknown error"}`,
-      );
+    return toBuyWithFiatStatus({ intentId: params.intentId, result });
+  } catch (error) {
+    // If the session is not found, the Onramp.status endpoint will return a 404 which we map to
```

Comment:
This should never happen since the onramps are stored on quote and if it does we should throw

Reply:
ok i wasn't sure. fixed